delivery_ups_oca/models/stock_picking.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def ups_get_label(self):
+        self.ensure_one()
+        tracking_ref = self.carrier_tracking_ref
+        if self.delivery_type != "ups" or not tracking_ref:
```

Comment:
if self.delivery_type != "ups" or not tracking_ref or not self.carrier_id:

Reply:
Carrier is always set, therefore this validation is not necessary.